spark/hbase-spark/src/main/scala/org/apache/hadoop/hbase/spark/DefaultSource.scala

```diff
@@ -150,12 +151,10 @@ case class HBaseRelation (
 
   def createTable() {
     val numReg = parameters.get(HBaseTableCatalog.newTable).map(x => x.toInt).getOrElse(0)
-    val startKey =  Bytes.toBytes(
-      parameters.get(HBaseTableCatalog.regionStart)
-        .getOrElse(HBaseTableCatalog.defaultRegionStart))
-    val endKey = Bytes.toBytes(
-      parameters.get(HBaseTableCatalog.regionEnd)
-        .getOrElse(HBaseTableCatalog.defaultRegionEnd))
+    val startKey = parameters.get(HBaseTableCatalog.regionStart)
+      .getOrElse(HBaseTableCatalog.defaultRegionStart).getBytes(StandardCharsets.ISO_8859_1)
```

Comment:
I'm not sure it is a good idea to use different encoding from the default used by `Bytes` util converter (StandardCharsets.UTF_8), as many pieces of hbase code would rely on the `Bytes` converter, comparisons may become inconsistent.

Also, I'm not sure I understand why you are using a different converter here, can you elaborate better what is the issue you are having within the builtin `Bytes` converter?

Reply:
The spark option use string to pass parameters, not support directly passing bytes，I need pass binary row key so I have to interpreter binary bytes as ISO_8859_1 encoded String, it’s not valid UTF-8.

It’s a trick, does break backward compatibility for UTF-8 string containing characters beyond ISO_8859_1 charset, the UTF-8 string must be wrapped as explained in the JIRA issue.

I can’t figure out better way to pass bytes in spark option.